rank_llm/retrieve_and_rerank.py

```diff
@@ -0,0 +1,170 @@
+from enum import Enum
+from typing import List, Union, Dict, Any
+from datetime import datetime
+import json
+from pathlib import Path
+
+from tqdm import tqdm
+
+from rank_llm.pyserini_retriever import PyseriniRetriever
+from rank_llm.rankllm import RankLLM
+
+
+class RetrievalMode(Enum):
+    DATASET = "dataset"
+    QUERY_AND_DOCUMENTS = "query_and_documents"
+    QUERY_AND_HITS = "query_and_hits"
+
+    def __str__(self):
+        return self.value
+    
+
+class Retriever:

```

Comment:
move Retriver and RetrievalMode to a different file. 
This retrieve_and_rerank.py should contain a function that implements the logic that is currently in `main() inside `rank_llm/run_rank_llm.py`. The latter can just get the flag values and call the retrive_and_rerank() function
So the final files will look like:
1-Retriver.py
2-Reranker.py
3-retrieve_and_rerank.py